feature/relayserver/relayserver_test.go

```diff
@@ -115,58 +124,78 @@ func Test_extension_profileStateChanged(t *testing.T) {
 	}
 }
 
-func Test_extension_handleBusLifetimeLocked(t *testing.T) {
+type mockRelayServer struct{}
+
+func (m mockRelayServer) Close() error { return nil }
+func (m mockRelayServer) AllocateEndpoint(_, _ key.DiscoPublic) (endpoint.ServerEndpoint, error) {
+	return endpoint.ServerEndpoint{}, errors.New("not implemented")
+}
+func (m mockRelayServer) GetSessions() []status.ServerSession { return nil }
+func (m mockRelayServer) SetDERPMapView(tailcfg.DERPMapView)  { return }
+
+type mockSafeBackend struct {
+	sys *tsd.System
+}
+
+func (m mockSafeBackend) Sys() *tsd.System         { return m.sys }
+func (m mockSafeBackend) Clock() tstime.Clock      { return nil }
+func (m mockSafeBackend) TailscaleVarRoot() string { return "" }
```

Comment:
(optional, mainly as a signal for the reader)
```suggestion
func (mockRelayServer) Close() error { return nil }
func (mockRelayServer) AllocateEndpoint(_, _ key.DiscoPublic) (endpoint.ServerEndpoint, error) {
	return endpoint.ServerEndpoint{}, errors.New("not implemented")
}
func (mockRelayServer) GetSessions() []status.ServerSession { return nil }
func (mockRelayServer) SetDERPMapView(tailcfg.DERPMapView)  {}

type mockSafeBackend struct {
	sys *tsd.System
}

func (m mockSafeBackend) Sys() *tsd.System         { return m.sys }
func (mockSafeBackend) Clock() tstime.Clock      { return nil }
func (mockSafeBackend) TailscaleVarRoot() string { return "" }
```